### PR TITLE
Enable unit testing for arm and remove the x64 shell support

### PR DESCRIPTION
**Pull request by swift-kim**
Part of https://github.com/flutter-tizen/engine/issues/239.

- Let the `flutter_tizen_unittests` target support all profiles by adding a necessary ldflag. This is now possible:<p>
  ```sh
  ninja -C out/linux_release_arm flutter_tizen_unittests
  ```
- Unsupport the x64 shell (the `flutter_tizen_shell` target and `flutter_tizen_shell.cc` file). x64 unit tests are still supported but any x64-specific code (`__X64_SHELL__`) will be removed in the future in favor of https://github.com/flutter-tizen/engine/issues/239.

### CODE REVIEW

File: shell/platform/tizen/BUILD.gn [hunk 1]
```diff
@@ -255,24 +255,13 @@ embedder("flutter_tizen_common") {
   defines = [ "COMMON_PROFILE" ]
 }
 
-embedder("flutter_desktop_source") {
+embedder("flutter_tizen_source") {
   target_type = "source_set"
   use_evas_gl_renderer = true
```
Comment (bbrto21) [inline shell/platform/tizen/BUILD.gn:260]: I think ecore_wl2 renderer should be preferred if we start to support `flutter_tizen_unittests` on all profile.
or we have to choose a renderer for each profile.

Comment (swift-kim) [reply shell/platform/tizen/BUILD.gn:260]: Okay, I'll check if it works.

Comment (swift-kim) [reply shell/platform/tizen/BUILD.gn:260]: The default value can be changed but doing so will break the existing CI unit tests. Let's apply this later in a separate PR.

**Review by bbrto21 — APPROVED** [review]

File: shell/platform/tizen/BUILD.gn (hunk 1 above)
Comment (bbrto21) [reply shell/platform/tizen/BUILD.gn:260]: Thanks for checking it out.

**Pull request merged**